lib/northbound_cli.h

```diff
@@ -57,7 +57,26 @@ extern void nb_cli_enqueue_change(struct vty *vty, const char *xpath,
 				  const char *value);
 
 /*
- * Apply enqueued changes to the candidate configuration.
+ * Apply enqueued changes to the candidate configuration, do not batch,
+ * and apply any pending commits along with the currently enqueued.
+ *
+ * vty
+ *    The vty context.
+ *
+ * xpath_base_fmt
+ *    Prepend the given XPath (absolute or relative) to all enqueued
+ *    configuration changes. This is an optional parameter.
+ *
+ * Returns:
+ *    CMD_SUCCESS on success, CMD_WARNING_CONFIG_FAILED otherwise.
+ */
+extern int nb_cli_apply_changes_clear_pending(struct vty *vty,
```

Comment:
Just to confirm my understanding. Will this function ensure older enqueued commands are going to be processed first and the current set of commands being enqueued are going to be attempted a short while later? If so which portion of code will ensure that the new set of commands are going to be eventually enqueued at a later point of time?

Reply:
It actually commits everything at once as everything has been queued at this point. They are applied in order as per normal. There will be no pending after this command.

Reply:
BTW, noticed that a slight change can be made in the "_internal_" function, it was calling `nb_cli_classic_commit` twice if `clear_pending` and `vty->pending_commit` was set.

Reply:
Yeah. Thanks for taking care of it.